evalml/data_checks/default_data_checks.py

```diff
@@ -19,11 +20,12 @@ class DefaultDataChecks(DataChecks):
         - `InvalidTargetDataCheck`
         - `NoVarianceDataCheck`
         - `ClassImbalanceDataCheck` (for classification problem types)
+        - `DateTimeNaNDataCheck`
 
     """
 
     _DEFAULT_DATA_CHECK_CLASSES = [HighlyNullDataCheck, IDColumnsDataCheck,
-                                   TargetLeakageDataCheck, InvalidTargetDataCheck, NoVarianceDataCheck]
+                                   TargetLeakageDataCheck, InvalidTargetDataCheck, NoVarianceDataCheck, DateTimeNaNDataCheck]
```

Comment:
I'm glad you added this here, do you think we could add a `DateTimeNanDataCheck` check for a default data checks test to make sure that these tests are running?